### ISSUE TITLE
imported bookmrks.html and noew no custom bookmarks

### ISSUE THREAD

**Issue by CSGalloway** [igorlogius/bookmark-branch-porter#5]
**Describe the bug**
I ha a lot of custom bookkmars and it would be great idea if before the bookmarks were imported - that a forced bookmark backup was done.  Now i have lost all my custom ones...

**To Reproduce**
1. Go to import as HTML
2. select "Other Bookmarks"
3. select bookmarks.html

**Expected behavior**
In caase of error - have a forced bookmarks backup done so they could be reimported by "Manage Bookkmarks" thne restore  or hwatver the top menu selection is.

**Desktop (please complete the following information):**
 - OS:  Windows 10
 - Browser FF
 - Version 115

**Smartphone (please complete the following information):**
 N/A

**Additional context**

**Comment by igorlogius** [comment]
>  Now i have lost all my custom ones...

Sorry if you experienced something unpleasent, because of my addon and thanks for taking the time to help improve it. 

>  it would be great idea if before the bookmarks were imported - that a forced bookmark backup was don

Well ... i could also just not remove existing bookmarks in the import folder.
Not 100% sure if that would align with user expectations, but i guess this would at least prevent accidental removal/loose of data. - Which i understand is pretty important.

**Comment by CSGalloway** [comment]
What did you end up doing pick us? I also received. I guess the newest source code for your ad on.Sent via the Samsung Galaxy S23 Ultra 5G, an AT&T 5G smartphone
-------- Original message --------From: igorlogius ***@***.***> Date: 7/7/23  5:04 AM  (GMT-05:00) To: igorlogius/bookmark-branch-porter ***@***.***> Cc: CSGalloway ***@***.***>, Author ***@***.***> Subject: Re: [igorlogius/bookmark-branch-porter] imported bookmrks.html and
  noew no custom bookmarks (Issue #5) 

Now i have lost all my custom ones...

Sorry if you experienced something unpleasent, because of my addon and thanks for taking the time to help improve it.

it would be great idea if before the bookmarks were imported - that a forced bookmark backup was don

Well ... i could also just not remove existing bookmarks in the import folder.
Not 100% sure if that would align with user expectations, but i guess this would at least prevent accidental removal/loose of data. - Which i understand is pretty important.

**Comment by igorlogius** [comment]
> What did you end up doing pick us? 

I am just not clearing  the import folder before you try to import something now. 
That way you should not loose any data. Hope that helps.

**Comment by CSGalloway** [comment]
Is there not a way to force a backup to:


%AppData%\Mozilla\Firefox\Profiles\i8mpx1mk.default-release\bookmarkbackups

folder?
===============
Charles S Galloway

**Comment by CSGalloway** [comment]
or would anytime the local bookmark file is written to, not read from, that
a copy of the bookmark file be copied to a subfolder of

%AppData%\Mozilla\Firefox\Profiles\i8mpx1mk.default-release\bookmark-branch-porter\bookmarkbackups

and maybe have it name the file YYYYMMDD_HHMMSS.html or .JSON or whatever
extension works the best?

===============
Charles S Galloway

**Comment by igorlogius** [comment]
closing since no accidnetal removal should now happend when importing something.
But if you want to further expand the issue, feel free to comment or open another ticket for it.

**Comment by CSGalloway** [comment]
Hi. What does "Assigned #5 to CSGallowa" mean?
===============
Charles S Galloway

**Comment by CSGalloway** [comment]
Is it not possible to check if there are any bookmark backups in (
C:\Users\[name]\AppData\Roaming\Mozilla\Firefox\Profiles\[profile
name]\bookmarkbackups ) ?
===============
Charles S Galloway

**Comment by igorlogius** [comment]
> Hi. What does "Assigned #5 to CSGallowa" mean?

dont be alarmed, i just like to assign issues if there are open questions i am waiting on.
but i also unassign when questions has been answered.

**Comment by igorlogius** [comment]
> Is it not possible to check if there are any bookmark backups in ( C:\Users\[name]\AppData\Roaming\Mozilla\Firefox\Profiles\[profile name]\bookmarkbackups ) ?

Sadly yes. It is AFAIK not possible for an addon to check this on its own.

**Comment by CSGalloway** [comment]
i will reach out to Firefox support later today and see what they say.....
===============
Charles S Galloway

**Comment by igorlogius** [comment]
> i will reach out to Firefox support later today and see what they say.....

Ok, ... good luck i guess. 

just fyi: This is the bookmark API 
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/bookmarks
which is available to addons. 

If you see/find anything that helps, feel free to comeback and we can discuss your findings.
